pkg/sources/postman/postman.go

```diff
@@ -0,0 +1,672 @@
+	if response.UID != "" {
+		m.Link = LINK_BASE_URL + "example/" + response.UID
+		m.FullID = response.UID
+	}
+	originalType := m.Type
+
+	if response.Header != nil {
+		vars := VariableData{
+			KeyValues: response.Header,
+		}
+		m.Type = originalType + " > response header"
+		s.scanVariableData(ctx, chunksChan, m, vars)
+	}
+
+	// Body in a response is just a string
+	if response.Body != "" {
+		m.Type = originalType + " > response body"
+		s.scanData(ctx, chunksChan, s.formatAndInjectKeywords(s.buildSubstitueSet(m, response.Body)), m)
+	}
+
+	if response.OriginalRequest.Method != "" {
+		m.Type = originalType + " > original request"
+		s.scanHTTPRequest(ctx, chunksChan, m, response.OriginalRequest)
+	}
+}
+
+func (s *Source) scanVariableData(ctx context.Context, chunksChan chan *sources.Chunk, m Metadata, variableData VariableData) {
+	if len(variableData.KeyValues) == 0 {
+		ctx.Logger().V(2).Info("no variables to scan", "type", m.Type, "uuid", m.FullID)
+		return
+	}
+
+	if !m.fromLocal && m.Type == COLLECTION_TYPE {
+		m.Link += "?tab=variables"
+	}
+
+	values := []string{}
+	for _, kv := range variableData.KeyValues {
+		s.addKeyword(kv.Key)
+		valStr := fmt.Sprintf("%v", kv.Value)
+		if valStr != "" {
+			s.sub.add(m, kv.Key, valStr)
+		} else if kv.SessionValue != "" {
+			valStr = fmt.Sprintf("%v", kv.SessionValue)
+		}
+		if valStr == "" {
+			continue
+		}
+		values = append(values, s.buildSubstitueSet(m, valStr)...)
+	}
+
+	m.FieldType = m.Type + " variables"
+	s.scanData(ctx, chunksChan, s.formatAndInjectKeywords(values), m)
+}
+
+func (s *Source) scanData(ctx context.Context, chunksChan chan *sources.Chunk, data string, metadata Metadata) {
+	if data == "" {
+		return
+	}
+	metadata.FieldType = metadata.Type
+
+	chunksChan <- &sources.Chunk{
+		SourceType: s.Type(),
+		SourceName: s.name,
+		SourceID:   s.SourceID(),
+		JobID:      s.JobID(),
+		Data:       []byte(data),
+		SourceMetadata: &source_metadatapb.MetaData{
+			Data: &source_metadatapb.MetaData_Postman{
+				Postman: &source_metadatapb.Postman{
+					Link:            metadata.Link,
+					WorkspaceUuid:   metadata.WorkspaceUUID,
+					WorkspaceName:   metadata.WorkspaceName,
+					CollectionId:    metadata.CollectionInfo.UID,
+					CollectionName:  metadata.CollectionInfo.Name,
+					EnvironmentId:   metadata.EnvironmentID,
+					EnvironmentName: metadata.EnvironmentName,
+					RequestId:       metadata.RequestID,
+					RequestName:     metadata.RequestName,
+					FolderId:        metadata.FolderID,
+					FolderName:      metadata.FolderName,
+					FieldType:       metadata.FieldType,
+					FieldName:       metadata.FieldName,
+					VariableType:    metadata.VarType,
+				},
+			},
+		},
+		Verify: s.verify,
+	}
+}
+
+func shouldSkip(uuid string, include []string, exclude []string) bool {
+	if slices.Contains(exclude, uuid) {
+		return true
+	}
+	if len(include) > 0 && !slices.Contains(include, uuid) {
+		return true
+	}
+	return false
+}
```

Comment:
Thanks for fixing this. I created this table to help me check the logic, so I'm going to leave it here in case it helps others too.

include list provided | included | excluded | shouldSkip |
|---|---|---|---|
| no | no | no | no |
| no | no | yes | yes |
| no | yes | no | X |
| no | yes | yes | X |
| yes | no | no | yes |
| yes | no | yes | yes |
| yes | yes | no | no |
| yes | yes | yes | yes |

`X` indicates impossible state (cannot have no provided include list and be included).